### PR TITLE
feat(benchmark): resolve issues #485-#492 for robust map benchmarking and CI signals

### LINKED ISSUE #485: Fix map benchmark resume semantics for multi-algorithm batches
## Context
Running map benchmark batches with multiple algorithms and resume enabled can skip later algorithms after the first algorithm writes resume state. In practice, `--no-resume` is currently needed to execute all algorithms reliably in one pass.

## Problem
Resume keys/state handling appear to be insufficiently scoped (likely not algorithm-aware), causing incorrect skip behavior and incomplete benchmark runs.

## Expected
Resuming should be safe and deterministic for multi-algorithm batches without requiring `--no-resume`.

## Tasks
- Reproduce with a minimal 2-algorithm map benchmark batch.
- Audit resume key derivation and persisted state shape.
- Ensure resume identity includes algorithm and relevant run dimensions.
- Add regression tests for mixed algorithm batches.
- Document intended resume semantics.

## Acceptance Criteria
- Multi-algorithm map benchmark runs complete correctly with resume enabled.
- No algorithm is skipped unless its own matching run unit is complete.
- Automated tests fail on old behavior and pass on fixed behavior.

### LINKED ISSUE #486: Make robot_sf_bench run fail fast on zero-episode / all-failed benchmark outcomes
## Context
`robot_sf_bench run` can report success even when map batch processing results in `0` written episodes and all jobs fail.

## Problem
Exit semantics and summary reporting do not reliably surface total run failure, which can mask invalid experiment results in CI and research pipelines.

## Expected
The CLI should return non-zero status when no valid episodes are produced (or when all scheduled jobs fail), and provide a clear failure summary.

## Tasks
- Define failure/success contract for benchmark execution.
- Track per-job and aggregate failure counts in final summary.
- Return non-zero exit code for all-failure / zero-episode outcomes.
- Add tests covering partial success vs full failure behavior.

## Acceptance Criteria
- Full-failure benchmark runs return non-zero exit code.
- Summary explicitly reports successful jobs, failed jobs, and written episodes.
- CI can gate on exit code without parsing logs.

### LINKED ISSUE #487: Define benchmark algorithm readiness tiers and enforce baseline-safe defaults
## Context
Classic benchmark sweeps show inconsistent baseline readiness:
- `orca`: partial success
- `goal`, `social_force`, `ppo`: poor success rates on quick subset
- `socnav_*`: hard failures in current default setup
- `sacadrl`: runtime failure
- `rvo/dwa/teb`: placeholder or unimplemented paths

## Problem
The benchmark currently mixes production-ready algorithms with partially integrated or placeholder implementations without explicit readiness signaling.

## Expected
Algorithm readiness should be explicit, and default benchmark suites should include only supported, validated baselines.

## Tasks
- Define readiness tiers (`experimental`, `baseline-ready`, `deprecated/placeholder`).
- Annotate algorithm registry with readiness metadata.
- Exclude or hard-fail placeholder algos in standard benchmark profiles.
- Add a benchmark preflight check that reports unsupported selections.
- Publish a canonical "paper baseline" algorithm set.

## Acceptance Criteria
- Running default benchmark profiles cannot silently include placeholder implementations.
- CLI reports readiness status before execution.
- Paper baseline config is versioned and reproducible.

### LINKED ISSUE #488: Harden SocNav planner benchmark integration with preflight checks and fallback policy
## Context
SocNav-based planners (`socnav_sampling`, `socnav_bench`) fail hard in benchmark runs when model/dependency expectations are not met.

## Problem
Current behavior provides weak guidance and can terminate runs instead of clean fallback/skip semantics.

## Expected
SocNav planner integration should have deterministic preflight validation, actionable errors, and explicit fallback policy.

## Tasks
- Add dependency/model artifact preflight checks before run start.
- Improve error messages with exact missing components and fix instructions.
- Implement configurable behavior: `fail-fast`, `skip-with-warning`, or `fallback`.
- Add integration tests for missing dependency and missing model cases.

## Acceptance Criteria
- Missing SocNav prerequisites are detected before simulation loop starts.
- Behavior under missing prerequisites is controlled by explicit CLI/preset policy.
- Errors are actionable and reproducible.

### LINKED ISSUE #489: Resolve SACADRL runtime indexing failure in classic benchmark scenarios
## Context
SACADRL benchmark execution currently hits a runtime indexing error (observed during classic subset sweeps).

## Problem
This blocks SACADRL from serving as a comparable baseline and can disrupt full benchmark batches.

## Expected
SACADRL runs should complete robustly on benchmark scenarios or be explicitly excluded with a clear reason.

## Tasks
- Reproduce failure with minimal scenario.
- Trace and fix indexing assumptions (state/action tensor or observation windowing).
- Add regression tests for affected scenario family.
- Re-enable SACADRL in baseline profile only after validation.

## Acceptance Criteria
- Previously failing SACADRL scenario executes without runtime indexing exceptions.
- Regression tests cover the root cause.
- Baseline profile status updated to reflect readiness.

### LINKED ISSUE #490: Establish paper-grade PPO baseline protocol for benchmark inclusion
## Context
PPO support was integrated into map benchmark execution, but quick subset evaluation shows very poor success rates.

## Problem
PPO is runnable but not benchmark-competitive yet, which risks misleading comparisons if included as-is.

## Expected
PPO baseline should have a documented, reproducible model provenance and minimum quality threshold before inclusion in headline benchmark tables.

## Tasks
- Define PPO evaluation protocol (checkpoint selection, normalization, deterministic seeds).
- Add model provenance metadata (training config, commit hash, dataset/version).
- Establish minimum acceptance metrics for baseline inclusion.
- Separate `experimental PPO` from `paper PPO baseline` profiles.

## Acceptance Criteria
- PPO benchmark entries include traceable model provenance.
- Benchmark profile clearly distinguishes experimental vs paper-grade PPO.
- Quality gate prevents accidental publication use of non-validated checkpoints.

### LINKED ISSUE #491: Stabilize force-related benchmark metrics against NaN/degenerate inputs
## Context
Metric post-processing has edge cases where force-related aggregates can become `NaN` (e.g., empty/all-invalid samples), reducing trust in downstream plots/tables.

## Problem
NaN leakage into benchmark artifacts can silently corrupt aggregate comparisons.

## Expected
Metrics should be numerically stable, with explicit handling for empty/invalid samples and transparent reporting.

## Tasks
- Audit force metric aggregation paths for empty slices and invalid values.
- Define policy: impute, drop-with-count, or mark metric unavailable.
- Add validation checks in report generation.
- Add tests covering empty/degenerate scenario outputs.

## Acceptance Criteria
- Benchmark metric tables no longer emit unhandled NaN values.
- Reports indicate sample counts and any filtered/invalid contributions.
- Degenerate cases are covered by tests.

### LINKED ISSUE #492: Add structured benchmark output and log-noise controls for CI/research pipelines
## Context
Benchmark runs emit noisy mixed logs (SDL/OpenCV/Pygame + planner logs), making automated parsing and CI diagnostics difficult.

## Problem
Operationally important benchmark signals (status/progress/failures) are not consistently separated from framework noise.

## Expected
Benchmark CLI should support structured, machine-readable output and configurable log verbosity.

## Tasks
- Add structured summary output option (e.g., JSON lines or final JSON blob).
- Route external library noise behind verbosity/debug flags where possible.
- Standardize error event formatting for parsers.
- Document recommended CI invocation.

## Acceptance Criteria
- CI can consume structured benchmark summaries without regex parsing.
- Default console output remains readable; verbose mode retains deep diagnostics.
- Failure diagnostics are consistently machine-detectable.

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Implement issues #485-#492 for map-benchmark reliability and CI usability: resume identity scoping, strict run exit semantics, algorithm readiness/profile gating, SocNav preflight policies, SACADRL robustness fixes, PPO paper-grade gating/provenance metadata, force-metric degenerate diagnostics, and structured CLI run outputs with log-noise controls.

## Linked issues
- Closes #485
- Closes #486
- Closes #487
- Closes #488
- Closes #489
- Closes #490
- Closes #491
- Closes #492

## Changes
- `#485`
- Map-runner resume identity no longer relies on `scenario_id--seed` only.
- Added map-specific episode identity with algorithm/run-dimension scoping and regression tests.

- `#486`
- `robot_sf_bench run` now returns non-zero when scheduled jobs produce zero written episodes.
- Resume no-op (`total_jobs==0`) remains success.

- `#487`
- Added `robot_sf/benchmark/algorithm_readiness.py` with readiness tiers.
- Added benchmark profiles: `baseline-safe` (default), `paper-baseline`, `experimental`.
- Placeholder planners are hard-blocked.
- Added canonical profile artifact: `configs/benchmarks/paper_baseline_algorithms_v1.yaml`.

- `#488`
- Added SocNav preflight + missing-prereq policy:
  - `fail-fast`
  - `skip-with-warning`
  - `fallback` (forces `allow_fallback=True` retry)

- `#489`
- Hardened SACADRL pedestrian payload normalization and heuristic path against malformed/empty arrays and count mismatches.

- `#490`
- Added PPO paper-grade gate validation (required provenance keys + numeric quality gate threshold).
- Added PPO metadata propagation for profile/provenance/gate status.

- `#491`
- Added `force_sample_stats` diagnostics to metrics output to report raw/finite/invalid force sample contributions and degenerate status.

- `#492`
- Added structured run events:
  - `--structured-output json`
  - `--structured-output jsonl`
- Added standardized event names (`benchmark.run.failure|summary|error`).
- Added external noise controls: `--external-log-noise auto|suppress|verbose`.

## Tests
- `uv run ruff check robot_sf/benchmark/algorithm_readiness.py robot_sf/benchmark/map_runner.py robot_sf/benchmark/runner.py robot_sf/benchmark/cli.py robot_sf/planner/socnav.py tests/benchmark/test_map_runner_preflight_profiles.py tests/test_socnav_sacadrl_adapter.py`
- `uv run pytest tests/benchmark/test_map_runner_resume_identity.py tests/test_cli_run_exit_semantics.py tests/benchmark/test_map_runner_utils.py -q`
- `uv run pytest tests/benchmark/test_map_runner_preflight_profiles.py tests/test_socnav_sacadrl_adapter.py tests/test_map_runner_ppo.py -q`
- `uv run pytest tests/unit/benchmark/test_force_sample_stats.py -q`
- `uv run pytest tests/test_cli_run_structured_output.py tests/test_cli_run_exit_semantics.py -q`
- `uv run pytest tests`
  - Result: `1779 passed, 10 skipped, 10 warnings` in `556.94s`

## Demo
- N/A (CLI/backend behavior and test coverage changes)

## Risks / rollout
- Map-runner resume IDs now include a hash suffix; existing older map episode IDs are not reused for skip matching (safe but may rerun older outputs once).
- `baseline-safe` is now default profile and blocks experimental algorithms unless explicitly enabled.

## Docs
- Updated: `docs/benchmark_spec.md`
- Added execution trace: `docs/context/issues_485_492_execution.md`
- Added profile artifact: `configs/benchmarks/paper_baseline_algorithms_v1.yaml`

## Validation
- Full suite run completed successfully after all commits.
- Per-issue regression tests added for resume identity, profile/preflight gating, SACADRL robustness, force diagnostics, and structured CLI output.

## Future Tasks
- Optional: extend structured events to baseline/aggregate/table commands for end-to-end pipeline parity.

### CODE REVIEW

File: robot_sf/benchmark/map_runner.py [hunk 1]
```diff
@@ -79,14 +196,126 @@ def _goal_policy(obs: dict[str, Any], *, max_speed: float = 1.0) -> tuple[float,
...
+        vel = ped_vel[idx] if idx < ped_vel.shape[0] else np.zeros(2, dtype=float)
+        agents.append(
+            {
+                "position": [float(ped_pos[idx, 0]), float(ped_pos[idx, 1])],
+                "velocity": [float(vel[0]), float(vel[1])],
+                "radius": ped_radius,
+            }
+        )
+
+    dt_raw = np.asarray(obs.get("dt", [0.1]), dtype=float).reshape(-1)
```
[the remark this answers is not in the document]
Comment (ll7) [reply robot_sf/benchmark/map_runner.py]: Implemented in df5cc73f: dt resolution now prefers `obs["sim"]["timestep"]` via `_extract_ppo_dt`, with fallback to legacy `obs["dt"]`/0.1. Covered by `test_obs_to_ppo_format_uses_ped_count_and_sim_timestep`.

File: robot_sf/benchmark/map_runner.py [hunk 2]
```diff
@@ -79,14 +196,126 @@ def _goal_policy(obs: dict[str, Any], *, max_speed: float = 1.0) -> tuple[float,
...
+    agents = []
+    for idx in range(ped_pos.shape[0]):
+        vel = ped_vel[idx] if idx < ped_vel.shape[0] else np.zeros(2, dtype=float)
+        agents.append(
+            {
+                "position": [float(ped_pos[idx, 0]), float(ped_pos[idx, 1])],
+                "velocity": [float(vel[0]), float(vel[1])],
+                "radius": ped_radius,
+            }
+        )
```
[the remark this answers is not in the document]
Comment (ll7) [reply robot_sf/benchmark/map_runner.py]: Implemented in df5cc73f: `_obs_to_ppo_format` now uses count-aware pedestrian extraction (`_extract_ppo_pedestrians`) and clamps to available rows, so padded ghost rows are not forwarded to PPO. Added regression test `test_obs_to_ppo_format_uses_ped_count_and_sim_timestep`.

File: docs/context/issues_485_492_execution.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- Every issue gets a separate commit and issue comment with commit reference.
+- Validation plan: targeted tests per issue, then full benchmark-related test run, then full test suite.
+- #485: map-runner now uses map-specific episode ids scoped by algorithm + run dimensions.
+- #486: CLI `run` now exits with non-zero when scheduled jobs produce zero written episodes.
+- #487: added versioned readiness catalog + profile gating (`baseline-safe`, `paper-baseline`, `experimental`).
+- #488: added SocNav preflight policy (`fail-fast`, `skip-with-warning`, `fallback`) before simulation loop.
+- #489: hardened SACADRL pedestrian payload normalization to avoid malformed indexing/axis errors.
+- #490: PPO paper profile now requires provenance fields + explicit quality-gate pass.
+- #491: added `force_sample_stats` diagnostics (raw/finite/invalid counts + status) in metrics output.
+- #492: added structured run events (`json`/`jsonl`) and external log-noise suppression controls.
```
[the remark this answers is not in the document]
Comment (ll7) [reply docs/context/issues_485_492_execution.md]: Implemented in df5cc73f: added an index entry in `docs/README.md` under Benchmarking & Metrics for `docs/context/issues_485_492_execution.md` so the doc is discoverable.
